javascript-carousel/main.js

```diff
@@ -0,0 +1,85 @@
+var $images = document.querySelectorAll('img');
+var $previousArrow = document.getElementById('previousImg');
+var $nextArrow = document.getElementById('nextImg');
+var $dots = document.querySelectorAll('i');
+var $progressDots = document.querySelector('.progress-dots');
+
+var start = null;
+var currentIndex = null;
+
+NodeList.prototype.indexOf = Array.prototype.indexOf;
```

Comment:
In general, we never modify existing prototypes unless it's to "polyfill" something that is a part of the specification but hasn't been implemented in browsers yet.